docs/nui-components/Button.md

```diff
@@ -0,0 +1,219 @@
+[HomePage](./Guide.md)<br>
+
+# Button
+This tutorial describes how to create and use button.
+
+## Overview
+Button is one kind of common component and clearly describes what action will occur when user select it.
+
+- Button can contain text or icon.
+- Button can be created by property or attributes.
+
+## Create with property
+1. Create Button by default constructor
+
+~~~{.cs}
+utilityBasicButton = new Button();
+~~~
+
+2. Set button property
+
+~~~{.cs}
+utilityBasicButton.IsSelectable = true;
+utilityBasicButton.BackgroundImageURL = "rectangle_btn_normal.png";
+utilityBasicButton.BackgroundImageBorder = new Rectangle(5, 5, 5, 5);
+utilityBasicButton.ShadowImageURL = "rectangle_btn_shadow.png";
+utilityBasicButton.ShadowImageBorder = new Rectangle(5, 5, 5, 5);
+utilityBasicButton.OverlayImageURLSelector = new StringSelector
+{
+    Pressed = "rectangle_btn_press_overlay.png",
+    Other = ""
+};
+utilityBasicButton.OverlayImageBorder = new Rectangle(5, 5, 5, 5);
+
+utilityBasicButton.TextColorSelector = new ColorSelector
+{
+    Normal = new Color(0, 0, 0, 1),
+    Pressed = new Color(0, 0, 0, 0.7f),
+    Selected = new Color(0.058f, 0.631f, 0.92f, 1),
+    Disabled = new Color(0, 0, 0, 0.4f)
+};
+
+utilityBasicButton.Size2D = new Size2D(300, 80);
+utilityBasicButton.Position2D = new Position2D(100, 300);
+utilityBasicButton.PointSize = 20;
+utilityBasicButton.Text = "UtilityBasicButton";
+root.Add(utilityBasicButton);
+~~~
+
+Button created by property:
+
+![ButtonProperty](../../assets/images/components/ButtonProperty.PNG)
+
+Button created by property pressed:
+
+![ButtonPropertyPressed](../../assets/images/components/ButtonPropertyPressed.PNG)
+## Create with attributes
+1. Create button attributes
+
+~~~{.cs}
+ButtonAttributes familyBasicButtonAttributes = new ButtonAttributes
```

Comment:
Its nice to have a detailed example of a button with many levels of images and 2 styles but first we should start with a simple button example.  A single background image and some text.  Later in the guide we can have this full example.